packages/next/client/components/noop-head.tsx

```diff
@@ -0,0 +1,10 @@
+import { warnOnce } from '../../shared/lib/utils/warn-once'
+
+export default function NoopHead() {
+  if (process.env.NODE_ENV !== 'production') {
```

Comment:
You can move this to the outside of the function, so the warning will be logged even if the component isn’t rendered (e.g. it can be behind a condition).